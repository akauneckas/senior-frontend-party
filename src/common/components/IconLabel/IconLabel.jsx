import React from 'react';
import PropTypes from 'prop-types';
import './IconLabel.scss';

const propTypes = {
    icon: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
    children: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};

const IconLabel = ({ icon, children, onClick }) => (
    <div className="IconLabel" onClick={onClick}>
        <img className="mr-2" alt="Logo" src={icon} />
        <span>{ children }</span>
    </div>
);

IconLabel.propTypes = propTypes;

export default IconLabel;
