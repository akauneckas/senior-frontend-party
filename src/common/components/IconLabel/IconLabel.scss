.IconLabel {
  display: flex;
  align-items: center;
  &:hover {
    cursor: pointer;
    color: #99cc33;
  }
}