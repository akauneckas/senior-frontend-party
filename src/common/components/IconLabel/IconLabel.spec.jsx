import React from 'react';
import renderer from 'react-test-renderer';
import IconLabel from './IconLabel';
import IconLogout from '../../../assets/icons/ico-logout.png';

describe('<IconLabel />', () => {
    it('should match snapshot', () => {
        const component = renderer.create(
            <IconLabel icon={IconLogout} onClick={() => {}}>Logout</IconLabel>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
