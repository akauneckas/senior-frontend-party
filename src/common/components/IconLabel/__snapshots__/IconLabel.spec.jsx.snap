// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconLabel /> should match snapshot 1`] = `
<div
  className="IconLabel"
  onClick={[Function]}
>
  <img
    alt="Logo"
    className="mr-2"
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <span>
    Logout
  </span>
</div>
`;
